File: dart/libmonet/lib/palettes/core_palette.dart
import 'package:libmonet/hct/cam16.dart';
import 'package:libmonet/hct/hct.dart';

import '../math.dart';
import '../utils.dart';
import 'tonal_palette.dart';

class CorePalette {
  final TonalPalette a1;
  final TonalPalette a2;
  final TonalPalette a3;
  final TonalPalette n1;
  final TonalPalette n2;
  final TonalPalette error;
  CorePalette._(this.a1, this.a2, this.a3, this.n1, this.n2, this.error);

  factory CorePalette(int argb) {
    final cam = CAM16.fromIntInViewingConditions(argb);
    var JSCompiler_inline_result =
        new HCT(cam.hue, cam.chroma, lstarFromInt(argb));
    final hue = JSCompiler_inline_result.hue;
    return CorePalette._(
      new TonalPalette(hue, Math.max(48, JSCompiler_inline_result.chroma)),
      new TonalPalette(hue, 16),
      new TonalPalette(hue + 60, 24),
      new TonalPalette(hue, 4),
      new TonalPalette(hue, 8),
      new TonalPalette(25, 84),
    );
  }
}

File: working_tree/google3/ux/material/libmonet/palettes/core_palette.js
'use strict';
const utils = require('../utils/color_utils/code.js');
const hct = require('../hct.js')
const tonalPalette = require('./tonal_palette.js')

module.exports = {
    CorePalette: CorePalette,
}
class CorePalette {
    constructor(argb) {
        const cam = hct.CAM16.fromIntInViewingConditions(argb);
        var JSCompiler_inline_result = new hct.HCT(cam.hue, cam.chroma, utils.lstarFromInt(argb));
        const hue = JSCompiler_inline_result.hue;
        this.a1 = new tonalPalette.TonalPalette(hue, Math.max(48, JSCompiler_inline_result.chroma));
        this.a2 = new tonalPalette.TonalPalette(hue, 16);
        this.a3 = new tonalPalette.TonalPalette(hue + 60, 24);
        this.n1 = new tonalPalette.TonalPalette(hue, 4);
        this.n2 = new tonalPalette.TonalPalette(hue, 8);
        this.error = new tonalPalette.TonalPalette(25, 84);
    }
};